src/components/molecules/tables/eventTable/eventTable.tsx

```diff
@@ -5,15 +5,81 @@ import { useHistory } from 'react-router-dom';
 import Icon from 'components/atoms/icons/icon';
 import styles from './eventTable.module.scss';
 import { transformDate } from 'utils/timeConverter';
+import { useEffect, useState } from 'react';
+import Modal from 'components/molecules/modal/Modal';
+import ConformationBox from 'components/molecules/confirmationBox/ConfirmationBox';
+import useConfirmation from 'hooks/useConfirmation';
+import { deleteEvent } from 'api';
+import { useToast } from 'hooks/useToast';
 
 const EventTable = () => {
-  const { events } = useUpcomingEvents();
+  const { events, error, setFetch } = useUpcomingEvents();
+  const [deleteModal, setOpenDeleteModal] = useState(false);
+  const [selectedEvent, setSelected] = useState<
+    Pick<Event, 'eid' | 'title'> | undefined
+  >();
   const history = useHistory();
+  const { addToast } = useToast();
 
   const moveToEventAdminPage = (eid: string) => {
     history.push(`event/${eid}/admin`);
   };
 
+  const openDeleteColumn = (eid: string) => {
+    const selected = events?.find((mem) => {
+      return mem.eid === eid;
+    });
+    setSelected(selected);
+    setOpenDeleteModal(true);
+  };
+
+  const adminDeleteEvent = async () => {
+    setOpenDeleteModal(true);
```

Comment:
Why is this needed? Isn't this already open when this is called?

Reply:
yes, kinda defeated the point of having it in a hook